Flowchart/wwwroot/js/React/Components/Canvas.jsx

```diff
@@ -6,89 +6,331 @@ class Canvas extends React.Component {
+            parentId:            -1,
+            newChildId:          -1
+        });
+    }
+
+    openEditStepModal(stepId) {
+        /** Open "Edit Step" Modal
+         *    Opens the modal form for editting a step
+         */
+        let titleText, descriptionText;
+        for (let step of this.state.stepList) {
+            if (step && step.id === stepId) {
+                titleText = step.title;
+                descriptionText = step.description;
+            }
+        }
+        this.setState({
+            editStepModalIsOpen:  true,
+            stepComponentList:    this.editOverlay(false),
+            editStepId:           stepId,
+            titleText:            titleText,
+            descriptionText:      descriptionText
+        });
+    }
+
+    closeEditStepModal() {
+        /** Close "Edit Step" Modal
+         *    Closes the modal form for editting a step
+         */
+        this.setState({
+            editStepModalIsOpen:  false,
+            descriptionText:      "",
+            stepComponentList:    this.editOverlay(true),
+            titleText:            "",
+            parentId:             -1,
+            newChildId:           -1,
+            editStepId:           -1
+        });
     }
 
-    addToStepList(newStep) {
-        let stepList = this.state.stepList;
-        stepList.push(newStep);
+    openDeleteStepModal(stepId) {
+        /** Open "Delete Step" Modal
+         *    Opens the modal form for deleting a step
+         */
         this.setState({
-            stepList: stepList,
-            body: stepList,
-            titleText: "",
-            descriptionText: ""
+            deleteStepModalIsOpen: true,
+            stepComponentList:      this.editOverlay(false),
+            deleteStepId:          stepId
         });
     }
 
-    addNewStep(event) {
+    closeDeleteStepModal() {
+        /** Close "Delete Step" Modal
+         *    Closes the modal form for deleting a step
+         */
+        this.setState({
+            deleteStepModalIsOpen: false,
+            stepComponentList:      this.editOverlay(true),
+            deleteStepId:          -1
+        });
+    }
+
+    /**************************************************************
+     * FORM SUBMISSION FUNCTIONS
+     *************************************************************/
+
+    editStep(event) {
         event.preventDefault();
-        let newStep = this.createStepComponent();
-        this.addToStepList(newStep);
-        this.closeModal();
+        let newStepList = this.state.stepList;
+        newStepList = newStepList.map((step) => {
+            if (step.id === this.state.editStepId) {
+                step.title = this.state.titleText;
+                step.description = this.state.descriptionText;
+            }
+            return step;
+        });
+        this.createComponentsFromStepList(newStepList);
+        this.closeEditStepModal();
+    }
+
+    deleteStep(event) {
```

Comment:
Ahhh I see how it all works together now. You delete the keys and recreate the graph from the new sets of child/parent relations. Neat.